ethcore/src/db.rs

```diff
@@ -218,11 +218,18 @@ impl Writable for DBTransaction {
 }
 
 impl<KVDB: KeyValueDB + ?Sized> Readable for KVDB {
-	fn read<T, R>(&self, col: Option<u32>, key: &Key<T, Target = R>) -> Option<T> where T: rlp::Decodable, R: Deref<Target = [u8]> {
+	fn read<T, R>(&self, col: Option<u32>, key: &Key<T, Target = R>) -> Option<T>
+		where T: rlp::Decodable, R: Deref<Target = [u8]> {
 		let result = self.get(col, &key.key());
 
 		match result {
-			Ok(option) => option.map(|v| rlp::decode(&v)),
+			Ok(option) => option.map(|v| match rlp::decode(&v){
+				Ok(decoded_rlp) => decoded_rlp,
+				//REVIEW: Should we panic here?
+				Err(decode_err) => {
+					panic!("decoding db value failed, key={:?}, err={}", &key.key() as &[u8], decode_err)
```

Comment:
@debris the situation here is similar to the above, db value read from disc, so panicking is ok. Q: do you prefer me to shorten this up with a `rlp::decode(&v).expect("decoding db value failed")` or is it useful to have the additional info about which variant of the decode error and the key?

Reply:
I believe it could be shortened with:

```rust
result.and_then(|option| match option {
  None => Ok(None),
  Some(v) => rlp::decode(&v).map(Some)
}).expect("decoding db value failed)
```